float64 cx
float64 cy
float64 area
bool appear
int16 mode
